concepts/05 Widgets/Chart/92 Bi-Directional Bar Chart/01 Prepare Data.md

```diff
@@ -69,4 +69,89 @@ To prepare data for the bi-directional bar chart, convert one of the two sets of
         [dataSource]="dataSource">
     </dx-chart>
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxChart ...
+            :data-source="dataSource">
+        </DxChart>
+    </template>
+
+    <script>
+    import DxChart from 'devextreme-vue/chart';
+    import DataSource from 'devextreme/data/data_source';
+    import ArrayStore from 'devextreme/data/array_store';
+
+    const population = [
+        { age: '0-4', male: 3.1, female: 2.9 },
+        { age: '5-9', male: 3.1, female: 3.0 },
+        // ...
+    ];
+
+    export default {
+        components: {
+            DxChart
+        },
+        data() {
+            return {
+                dataSource: new DataSource({
+                    store: new ArrayStore({
+                        data: population
+                    }),
+                    map: (dataItem) => {
+                        return {
+                            age: dataItem.age,
+                            male: dataItem.male,
+                            female: -dataItem.female // Changing the values' sign
+                        }
+                    }
+                })
+            };
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import Chart from 'devextreme-react/chart';
+    import DataSource from 'devextreme/data/data_source';
+    import ArrayStore from 'devextreme/data/array_store';
+
+    const population = [
+        { age: '0-4', male: 3.1, female: 2.9 },
+        { age: '5-9', male: 3.1, female: 3.0 },
+        // ...
+    ];
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+            this.state = {
+                dataSource: new DataSource({
```

Comment:
There's no reason to put the data source in the state.

Reply:
Fixed